lib/rules/indent.js

```diff
@@ -807,6 +889,71 @@ module.exports = {
             });
         }
 
+        /**
+        * Ignore first token of each line within an unknown node if its
+        * offset does not depend on another token's offset. If the first token
+        * of a line's offset depends on the offset of another token, make the
+        * last dependent offset depend on the first token of its line.
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function ignoreUnknownNode(node) {
+            const startLine = node.loc.start.line;
+            const endLine = node.loc.end.line;
+
+            for (let lineNumber = startLine; lineNumber <= endLine; lineNumber++) {
+                const firstTokenOfLine = tokenInfo.firstTokensByLineNumber.get(lineNumber);
+
+                if (!firstTokenOfLine) {
+                    continue;
+                }
+
+                const tokenInNodeRange = (node.range[0] <= firstTokenOfLine.range[0]) &&
+                    (node.range[1] >= firstTokenOfLine.range[1]);
+
+                if (!tokenInNodeRange) {
+                    continue;
+                }
+
+                const dependency = offsets.getFirstDependency(firstTokenOfLine);
+
+                if (dependency) {
+                    let currDependency = dependency,
+                        lastDependency = null;
+
+                    while (
+                        (currDependency = offsets.getFirstDependency(currDependency)) &&
+                        currDependency.loc.start.line > startLine
```

Comment:
I am still concerned about iterating through dependencies, as this really isn't what the offset dependency tree is designed for. The main intended usage is to just set offsets and let the tree handle how tokens depend on each other. This allows nodes to behave consistently regardless of what types of parents they have.

Is this loop still necessary? I'm not sure I understand how the `offsets.matchIndentOf(firstTokenOfDependentLine, lastDependency);` on line 934 helps here.